src/Type/Definition/HasFieldsTypeImpl.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+declare(strict_types=1);
+
+namespace GraphQL\Type\Definition;
+
+use GraphQL\Error\InvariantViolation;
+use GraphQL\Utils\Utils;
+
+use function array_keys;
+
+trait HasFieldsTypeImpl
```

Comment:
Personally I'd rather design it as 

`class InterfaceType extends TypeWithFields`

`abstract class TypeWithFields extends Type implements HasFieldsType`

1. Traits often cause issues for static analyser tools and IDEs.
2. Also it would allow us to use a better name than `Impl`.
3. We could safely use `@inheritDoc` instead of duplicating phpdoc.

Though it's also a matter of preference I guess.

Reply:
I'm fine with this, it's a preference thing indeed, I'll make the adjustments 👍